apps/computeVSD/beerLambertProjectionImageFilter.h

```diff
@@ -0,0 +1,156 @@
+
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ *                          Jafet.VillafrancaDiaz@epfl.ch
+ *
+ * This file is part of Fivox <https://github.com/BlueBrain/Fivox>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef FIVOX_BEERLAMBERTPROJECTIONIMAGEFILTER_H
+#define FIVOX_BEERLAMBERTPROJECTIONIMAGEFILTER_H
+
+#include <itkProjectionImageFilter.h>
+#include <itkConceptChecking.h>
+
+#include <fivox/types.h>
+
+/**
+ * Filter that uses a Beer-Lambert approach to accumulate and project the
+ * values of the input image (e.g. 3D volume) as an output image
+ * (e.g. 2D image). Based on itkStandardDeviationProjectionImageFilter.h
+ */
+namespace Functor
+{
+template< typename TInputPixel, typename TAccumulate >
+class BeerLambertAccumulator
+{
+
+public:
+    typedef typename itk::NumericTraits< TInputPixel >::RealType RealType;
+
+    explicit BeerLambertAccumulator( itk::SizeValueType size )
+    {
+        m_Size = size;
+        m_Values.reserve( size );
+    }
+
+    ~BeerLambertAccumulator(){}
+
+    inline void Initialize()
+    {
+        m_Sum = itk::NumericTraits< TAccumulate >::ZeroValue();
+        m_Values.clear();
+    }
+
+    inline void operator()( const TInputPixel& input )
+    {
+        // assume that values are being computed in order (top-down)
+        const double depth =
+                m_PixelSize * ( m_Size - m_Values.size( )) / (double)m_Size;
```

Comment:
This computation yields values between 0 and m_PixelSize, but not between 0 and the volume depth in Y.